lib/cloud_controller/diego/stager_client.rb

```diff
@@ -76,6 +76,12 @@ def http_client
       def logger
         @logger ||= Steno.logger('cc.stager.client')
       end
+
+      def error_message(response)
+        JSON.parse(response.body)['error']['message'] || response.code
+      rescue JSON::ParserError, NoMethodError
```

Comment:
Are you rescuing NoMethodError to handle the case when the request doesn't have an "error" key? This will hide exceptions that occur when other objects are nil unexpectedly. It would be better to deal with the problem directly. Here is one way you could do that...

```ruby
JSON.parse(response.body).fetch('error', {})['message'] || response.code
```